### PR TITLE
HPCC-25980 update2 Docker image with TensorFlow 2.5

### PR DESCRIPTION
**Pull request by xwang2713**
Build platform-gnn and platform-gnn-gpu Docker images with TensorFlow 2.0 and latest Nvidia CUDA drivers

Signed-off by xiaoming wang <xiaoming.wang@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x ] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Build platform-gnn and platform-gnn-gpu locally

### CODE REVIEW

File: dockerfiles/platform-gnn-gpu/Dockerfile [hunk 1]
```diff
@@ -20,43 +20,51 @@
 # Others may not be wanted at all in container mode - tensoflow and nvidia  example??
 
 ARG BUILD_LABEL
-ARG DOCKER_REPO
-FROM ${DOCKER_REPO}/platform-core:${BUILD_LABEL}
+FROM hpccsystems/platform-core:${BUILD_LABEL}
```
Comment (Michael-Gardner) [inline dockerfiles/platform-gnn-gpu/Dockerfile:23]: What's the reasoning behind removing DOCKER_REPO and allowing us to target a custom platfom-core build?

File: dockerfiles/platform-gnn-gpu/Dockerfile [hunk 2]
```diff
@@ -66,37 +74,43 @@ RUN pip3 install       \
...
-ARG LIBNVINFER=6.0.1-1
-ARG LIBNVINFER_MAJOR_VERSION=6
+ARG LIBNVINFER=8.0.0-1
+ARG LIBNVINFER_MAJOR_VERSION=8
 
 # Needed for string substitution
 SHELL ["/bin/bash", "-c"]
 # Pick up some TF dependencies
-RUN apt-get update -y && apt-get install -y --no-install-recommends \
+RUN apt-get update && apt-get install -y --no-install-recommends \
```
Comment (Michael-Gardner) [inline dockerfiles/platform-gnn-gpu/Dockerfile:88]: losing the -y here, any chance we're going to run into a prompt stopping the command from executing?

**Comment by richardkchapman** [comment]
Looks like you are targeting the wrong version and/or based on the wrong branch

**Comment by xwang2713** [comment]
Changed base to candidate-8.2.0

File: dockerfiles/platform-gnn-gpu/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+When preparing the Dockerfile, specially for Tensorflow 2 with GPU support it is import to reference
+Tensorflow and Nvidia Cuda Dockerfiles and Docker images to pick compatible libabries.
+
+### Docker image and Dockerfile Reference
+
+#### Tensforflow
+
+- Dockerfile:
+  https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/dockerfiles/dockerfiles/gpu.Dockerfile
+  Above Dockeerfile is based https://hub.docker.com/layers/nvidia/cuda/. Here are runime, base, etc.
```
Comment (Michael-Gardner) [inline dockerfiles/platform-gnn-gpu/README.md:16]: dockeerfile -> dockerfile,  runime -> runtime?

File: dockerfiles/platform-gnn-gpu/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# HPCC Systems Machine Learning GNN GPU Docker Images
+
+## Tensorflow 2 with GPU support
+
+The new Tensorflow 2 and Nvidia Cuda libraries are not always compatible. It will be ideal if we can use Docker image by published by Tensorflow (Docker Hub tensorflow/tensorflow <version>-gpu) but the base image is Ubuntu 18.04 and we use Ubuntu 20.04. So instead we create the images by addiing Tensorflow 2 and Nvida Cuda, etc libraries on the top of hpccystems/platform-core image.
+
+When preparing the Dockerfile, specially for Tensorflow 2 with GPU support it is import to reference
```
Comment (Michael-Gardner) [inline dockerfiles/platform-gnn-gpu/README.md:7]: import -> important

**Review by Michael-Gardner — CHANGES_REQUESTED** [review]
some small changes and clarifications. Otherwise looks great.

**Review by xwang2713 — COMMENTED** [review]
Need change "hpccsystems" to ${DOCKER_REPO}

File: dockerfiles/platform-gnn-gpu/Dockerfile (hunk 1 above)
Comment (xwang2713) [reply dockerfiles/platform-gnn-gpu/Dockerfile:23]: fixed

File: dockerfiles/platform-gnn-gpu/Dockerfile (hunk 2 above)
Comment (xwang2713) [reply dockerfiles/platform-gnn-gpu/Dockerfile:88]: fixed

File: dockerfiles/platform-gnn-gpu/README.md (hunk 3 above)
Comment (xwang2713) [reply dockerfiles/platform-gnn-gpu/README.md:16]: I mean "There are". updated

File: dockerfiles/platform-gnn-gpu/README.md (hunk 4 above)
Comment (xwang2713) [reply dockerfiles/platform-gnn-gpu/README.md:7]: fixed

**Comment by xwang2713** [comment]
Fixed both gnn/Dockerfile and gnn-gpu/Dockerfile

**Comment by xwang2713** [comment]
Updated and rebased

File: dockerfiles/platform-gnn-gpu/Dockerfile [hunk 5]
```diff
@@ -20,43 +20,51 @@
 # Others may not be wanted at all in container mode - tensoflow and nvidia  example??
 
 ARG BUILD_LABEL
-ARG DOCKER_REPO
 FROM ${DOCKER_REPO}/platform-core:${BUILD_LABEL}
```
Comment (Michael-Gardner) [inline dockerfiles/platform-gnn-gpu/Dockerfile:23]: Do we need to have 
ARG DOCKER_REPO

before using the variable here?

File: dockerfiles/platform-gnn/Dockerfile [hunk 6]
```diff
@@ -20,11 +20,10 @@
 # Others may not be wanted at all in container mode - tensorflow, etc  ??
 
 ARG BUILD_LABEL
-ARG DOCKER_REPO
 FROM ${DOCKER_REPO}/platform-core:${BUILD_LABEL}
```
Comment (Michael-Gardner) [inline dockerfiles/platform-gnn/Dockerfile:23]: Do we need to have 
ARG DOCKER_REPO

before using the variable here?

**Review by Michael-Gardner — CHANGES_REQUESTED** [review]

File: dockerfiles/platform-gnn/Dockerfile (hunk 6 above)
Comment (xwang2713) [reply dockerfiles/platform-gnn/Dockerfile:24]: fixed

File: dockerfiles/platform-gnn-gpu/Dockerfile (hunk 5 above)
Comment (xwang2713) [reply dockerfiles/platform-gnn-gpu/Dockerfile:24]: fixed

**Review by Michael-Gardner — APPROVED** [review]
@xwang2713 looks good

**Pull request merged**